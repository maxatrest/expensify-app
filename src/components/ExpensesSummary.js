import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component {
    render() {
        return (
            <div>
                <p>Viewing {this.props.expenses.length} expense{(this.props.expenses.length > 1 || this.props.expenses.length === 0) ? 's' : null} totalling {
                    numeral(expensesTotal(this.props.expenses)/100).format('$0,0.00')
                }</p>
            </div>
        )

    }
}

const mapStateToProps = (state) => ({expenses: selectExpenses(state.expenses, state.filters)})

export default connect(mapStateToProps)(ExpensesSummary)