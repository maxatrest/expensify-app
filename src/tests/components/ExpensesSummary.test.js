import React from 'react'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'
import { shallow } from 'enzyme'

test('Should capture Expenses Summary when only one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should capture Expenses Summary when multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()
})
