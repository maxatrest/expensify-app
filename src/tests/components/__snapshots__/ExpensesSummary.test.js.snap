// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should capture Expenses Summary when multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expense
    s
     totalling 
    $1,141.95
  </p>
</div>
`;

exports[`Should capture Expenses Summary when only one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
     totalling 
    $1.95
  </p>
</div>
`;
